cpu/atmega_common/cpu.c

```diff
@@ -1,33 +1,50 @@
 /*
- * Copyright (C) 2017 RWTH Aachen, Josua Arndt
+ * Copyright (C) 2014 Freie Universität Berlin, Hinnerk van Bruinehsen
+ *               2017 RWTH Aachen, Josua Arndt
+ *               2018 Matthew Blue
  *
  * This file is subject to the terms and conditions of the GNU Lesser
  * General Public License v2.1. See the file LICENSE in the top level
  * directory for more details.
  */
 
 /**
- * @ingroup     cpu_atmega256rfr2
+ * @ingroup     cpu_atmega_common
  * @{
  *
  * @file
  * @brief       Implementation of the CPU initialization
  *
+ * @author      Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
  * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
  * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+
  * @}
  */
 
 #include <avr/io.h>
 #include <avr/wdt.h>
 #include <avr/pgmspace.h>
+
 #include "cpu.h"
 #include "board.h"
 #include "periph/init.h"
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+#if !defined (CPU_ATMEGA256RFR2)
```

Comment:
It's a bit difficult because it would clash with cpu.c from atmega_common. The options are this `#if !defined` or copy all the code in atmega_common to atmega256rfr2. Do you have an idea on how to avoid the clash?

Reply:
What about using something similar to the header guards:

```
#ifndef ATMEGA_CPU_C
#define ATMEGA_CPU_C

/* Code here */

#endif /* ATMEGA_CPU_C */
```

Reply:
Hmmm either I don't understand what you mean or I don't know how those guards work, but it's not that thing to avoid concurrency in a more "first come, first served" way? I'd need to investigate if the cpu.c from atmega256rfr2 comes first, so then it might work. Though, it's not common to do that in RIOT...

Reply:
I think a better way to do this would be to move all of the specific code into a function that is called by cpu_init. Sort of like how the LED code was refactored for the ATmega-based boards.